File: ImmoMange-ui/src/app/services/models/rental-history.ts
/* tslint:disable */
/* eslint-disable */
import { Property } from '../models/property';
import { Tenant } from '../models/tenant';
export interface RentalHistory {
  endDate?: string;
  id?: number;
  property?: Property;
  reasonForLeaving?: string;
  startDate?: string;
  tenant?: Tenant;
  wasEvicted?: boolean;
}

File: src/main/java/org/asm/immomanage/models/RentalHistory.java
package org.asm.immomanage.models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.time.LocalDate;
@Data
@RequiredArgsConstructor
@Entity
public class RentalHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne
    @JoinColumn(name = "property_id", nullable = false)
    private Property property;

    @OneToOne
    @JoinColumn(name = "tenant_id", nullable = false)
    private Tenant tenant;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalDate endDate;

    @Column(nullable = false)
    private boolean wasEvicted;

    @Column(nullable = false)
    private String reasonForLeaving;

}
